Sources/Core/Presenter/UploadingItem.swift

```diff
@@ -156,71 +135,44 @@ public final class UploaderItem: Equatable {
         extraData = nil
     }
     
-    private func createUploading() -> Observable<ProgressResponse<URL>> {
-        guard let channel = channel else {
-            return .empty()
-        }
-        
-        let request: Observable<ProgressResponse<URL>>
-        
+    private func encodeData() {
         if type == .file || type == .video {
             if let url = url, let data = try? Data(contentsOf: url) {
-                request = channel.rx.sendFile(fileName: fileName, mimeType: fileType.mimeType, fileData: data)
+                self.mimeType = fileType.mimeType
+                self.data = data
             } else {
-                return .error(ClientError.emptyBody(description: "For file at URL: \(url?.absoluteString ?? "<unknown>")"))
+                error = .emptyBody(description: "For file at URL: \(url?.absoluteString ?? "<unknown>")")
             }
-        } else {
-            let imageData: Data
-            var mimeType: String = fileType.mimeType
             
-            if let gifData = gifData {
-                imageData = gifData
-                mimeType = AttachmentFileType.gif.mimeType
-            } else if let url = url, let localImageData = try? Data(contentsOf: url) {
-                imageData = localImageData
-            } else if let encodedImageData = image?.jpegData(compressionQuality: 0.9) {
-                imageData = encodedImageData
-                mimeType = AttachmentFileType.jpeg.mimeType
-            } else {
-                let errorDescription = "For image: gifData = \(gifData == nil ? "no" : "yes"), "
-                    + "URL = \(url?.absoluteString ?? "<none>"), "
-                    + "image data: \(image?.description ?? "<none>")"
-                
-                return .error(ClientError.emptyBody(description: errorDescription))
-            }
+            return
```

Comment:
Is this `return` correct? If so, we should probably rewrite the whole `if` into `guard`.